### DISCUSSION TITLE
Tracking loading time of useQuery

### DISCUSSION THREAD

**Discussion by cubecleveland** [TanStack/query#4389] [Q&A]
Hello,

I have a use case of running unstable queries that can last more than 10 seconds.
I wan the query to cahnge the state or dipacth another state lets say after 5 seconds of loading so the ui can show to user that "Im still fetching..." and also possiblyy cancel the query after 10 seconds.....

IS there a boilerplate we might be able to use to achive that ?

**Comment by TkDodo** [comment]
Not that I'm aware of, no. You would have to to that with your data fetching function that you call from the queryFn.
